serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -114,8 +112,13 @@ def add(self, request_states: list[RequestState]):
                 if (
                     request_state.validation_err is not None
                     or request_state.prompt_len >= self.max_context_length
+                    or request_state.prompt_len >= self.max_num_batched_tokens > 0
+                    or (self.cache_manager.get_kv_cache_size() - request_state.prompt_len) < 0
```

Comment:
Why do we need this? At this point we know that `prompt_len < max_num_batched_tokens`. So we can process this request, after evicting the cache if necessary.

Reply:
we do not need the first condition (will remove), but we still need the second one. It is a verification that this request ill have a chance to appear in the current_batch. It relate to the condition with `prompt_allocate_ratio` that I substituted in `_adjust_batch`. If we do not have such verification in this place, we can have a situation when we will not be able to add such reqquest to current batch and this one of the use case when server hang could appear